appinventor/components/src/com/google/appinventor/components/runtime/Sharing.java

```diff
@@ -103,8 +109,13 @@ public void ShareFile(String file) {
       + "installed on the phone by displaying a list of available apps and allowing the user to " +
       " choose one from the list. The selected app will open with the file and message inserted on it.")
   public void ShareFileWithMessage(String file, String message) {
-    if (!file.startsWith("file://"))
-      file = "file://" + file;
+    if (!file.startsWith("file://")) {
+      if (!file.startsWith("/")) {
+        file = form.getFilesDir().getPath() + "/" +  file;
```

Comment:
Note that this logic will only work in compiled apps. The companion puts the private data files into a different directory for debugging purposes. Consider adding a method to `Form` called `getPrivateDataDir()` that we can use. Have `Form` return `getFilesDir()` and have `ReplForm` return the result of `QUtil.getReplDataPath` so that the logic is consistent with `File.AbsoluteFileName`.

Reply:
@ArjunInventor This is the only outstanding item on this PR. Do you have time to address it?

Reply:
Hey, @ewpatton , I'll address this over the weekend...classes + part-time has been a little hectic. Hope that's fine.